apps/performance-tests/src/StatelessHierarchyProvider.ts

```diff
@@ -0,0 +1,135 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+/* eslint-disable @itwin/no-internal */
+
+import { expand, filter, from, mergeAll, of, tap } from "rxjs";
+import { IModelDb } from "@itwin/core-backend";
+import { DbQueryRequest, DbQueryResponse, DbRequestExecutor, DbRequestKind, ECSqlReader } from "@itwin/core-common";
+import { ISchemaLocater, Schema, SchemaContext, SchemaInfo, SchemaKey, SchemaMatchType } from "@itwin/ecschema-metadata";
+import { createECSqlQueryExecutor, createMetadataProvider } from "@itwin/presentation-core-interop";
+import { createLimitingECSqlQueryExecutor, HierarchyNode, HierarchyProvider } from "@itwin/presentation-hierarchy-builder";
+import { ModelsTreeDefinition } from "@itwin/presentation-models-tree";
+
+const ENABLE_SCHEDULED_NODES_LOGGING = false;
+const ECSQL_ROW_LIMIT = 1000;
+
+export class StatelessHierarchyProvider {
+  private readonly _provider: HierarchyProvider;
+
+  constructor(
+    iModelDb: IModelDb,
+    private readonly _nodeRequestLimit = 10,
+  ) {
+    this._provider = createProvider(iModelDb);
+  }
+
+  public async loadInitialHierarchy(): Promise<void> {
+    await this.loadNodes(StatelessHierarchyProvider.initialHasChildren);
+  }
+
+  public async loadFullHierarchy(): Promise<void> {
+    await this.loadNodes(StatelessHierarchyProvider.fullHasChildren);
+  }
+
+  private static initialHasChildren(node: HierarchyNode): boolean {
+    return StatelessHierarchyProvider.fullHasChildren(node) && !!node.autoExpand;
+  }
+
+  private static fullHasChildren(node: HierarchyNode): boolean {
+    return node.children === true || (Array.isArray(node.children) && node.children.length > 0);
+  }
+
+  private async loadNodes(nodeHasChildren: (node: HierarchyNode) => boolean) {
+    let nodesCreated = 0;
+    let nodesScheduled = 0;
+    const timer = ENABLE_SCHEDULED_NODES_LOGGING ? setInterval(() => nodesScheduled && console.log(`Nodes scheduled ${nodesScheduled}`), 1000) : undefined;
```

Comment:
Do we need this?

Reply:
I am not sure, but it can be useful to easily determine if there's progress or not